src/evaluation/evaluator.h

```diff
@@ -483,7 +478,7 @@ class Searcher {
 
             undoMove(moveRes->hash);
 
-            if (s_isStopped.load(std::memory_order_relaxed) || m_searchState.load(std::memory_order_relaxed) == SearchState::Kill)
+            if (s_isStopped.load(std::memory_order_relaxed) || s_oneFullSearchComplete.load(std::memory_order_relaxed) || m_searchState.load(std::memory_order_relaxed) == SearchState::Kill)
```

Comment:
This is racy :racing_car: 
Also no need to store 3 atomics when we can just add the stopped / done search to the state :)